src/library/ImageProvider.cs

```diff
@@ -62,41 +70,53 @@ internal static class DevHostRestorationJob
         {
             // To change RestorationJobImageName tag, please update deployment\settings\services\imagetag.setting accordingly
             // During development, use environment variable LPK_RESTORATIONJOBIMAGENAME to override the default restorationjob image name
-            private static Lazy<string> _tag = new Lazy<string>(() => EmbeddedFileUtilities.GetImageTag("MINDARO_DEVHOSTAGENT_RESTORATIONJOB_TAG"));
+            private static Lazy<string> _tag = new(() => EmbeddedFileUtilities.GetImageTag("MINDARO_DEVHOSTAGENT_RESTORATIONJOB_TAG"));
 
             public static string Version => _tag.Value;
 
-            internal static string Name => $"lpkrestorationjob:{_tag.Value}";
+            internal static string Name => Common.Constants.ImageName.RestorationJobImageName;
         }
 
         private static class RoutingManager
         {
-            internal static string Name => $"{Common.Constants.Routing.RoutingManagerNameLower}:stable";
+            public static string Version => Common.Constants.ImageTag.RoutingManagerImageTag;
+            internal static string Name => Common.Constants.Routing.RoutingManagerNameLower;
         }
 
-        public ImageProvider(ILog log, IEnvironmentVariables environmentVariables)
+        public ImageProvider(ILog log, IEnvironmentVariables environmentVariables, IManagementClientFactory managementClientFactory)
         {
             _log = log;
             _environmentVariables = environmentVariables;
-
-            _devHostImage = new Lazy<string>(() => GetImage(_environmentVariables.DevHostImageName, DevHost.Name));
-            _devHostRestorationJobImage = new Lazy<string>(() => GetImage(_environmentVariables.DevHostRestorationJobImageName, DevHostRestorationJob.Name));
-            _routingManagerImage = new Lazy<string>(() => GetImage(_environmentVariables.RoutingManagerImageName, RoutingManager.Name));
+            var kubeConfigManagementClient = managementClientFactory.CreateKubeConfigClient();

```

Comment:
KubernetesClient is required to ListNodes in the Cluster to find out if node architecture is amd64 or arm64.